roles/utils_build_push_image/tasks/main.yml

```diff
@@ -1,169 +1,47 @@
-    delay: 30
-
-  - name: Fail if the image failed to be built
-    when: "'Failed' in wait_img_build.stdout"
-    fail: msg="The image failed to build"
-
-  always:
-  - name: Get BuildConfig description (debug)
-    shell: oc describe --namespace=utils-ci bc/{{ local_image_name }} > {{ artifact_extra_logs_dir }}/image-build.descr
-    failed_when: false
-
-  - name: Store the logs of image build (debug)
-    shell:
-      oc logs {{ build_name_cmd.stdout }} -n utils-ci > {{ artifact_extra_logs_dir }}/image-build.log
-    failed_when: false
-
-- name: Find and store the builder-dockercfg secret
-  block:
-  - name: Find the name of the builder-dockercfg secret
-    shell:
-      set -o pipefail;
-      oc get secrets -oname -n utils-ci | cut -d/ -f2 | grep builder-dockercfg
-    register: builder_secret_cmd
-
-  - name: Store the name of the secret in a variable
-    set_fact:
-      builder_secret: "{{ builder_secret_cmd.stdout }}"
-  rescue:
-  - name: Failed to find the builder-dockercfg secret
-    fail: msg="No builder-dockercfg secret in the 'utils-ci' namespace ..."
-
-- name: Get quay.io secret name
-  command:
-    yq -r .metadata.name "{{ auth_file }}"
-  register: quay_push_secret_name_cmd
-
-- name: Store quay.io secret name
-  set_fact:
-    quay_push_secret_name: "{{ quay_push_secret_name_cmd.stdout }}"
-
-- name: Create quay.io secret
-  command:
-    oc apply -f "{{ auth_file }}" -n utils-ci
-
-- name: Apply the helper image script manifest
+- name: Create namespace for images if does not exist
   shell:
     set -o pipefail;
-    oc create cm helper-image-script
-       -n utils-ci
-       "--from-file=run-helper-image-push.sh={{ utils_helper_image_push_script }}"
-       --dry-run -oyaml
-       | oc apply -f-
-
-- name: Instantiate the template of the image push pod
-  template:
-    src: "{{ utils_helper_image_push_pod }}"
-    dest: "{{ artifact_extra_logs_dir }}/helper-image-push-pod.yml"
-    mode: 0400
+    oc create namespace "{{ utils_build_push_image_namespace }}"
+       --dry-run=client
+       -oyaml
+    | tee "{{ artifact_extra_logs_dir }}/000_namespace.yml"
+    | oc apply -f-
 
-- name: Delete image push pod if already exists
-  command: oc delete pod {{ local_image_name }}-push-pod -n utils-ci
+- name: Create the imagestream, if does not exist
+  shell:
+    set -o pipefail &&
+    oc create imagestream "{{ utils_build_push_image_local_name }}"
+       --dry-run=client
+       -oyaml
+    | tee "{{ artifact_extra_logs_dir }}/001_imagestream.yml"
+    | oc apply -f- -n "{{ utils_build_push_image_namespace }}"
+
+- name: Check if the image is already built
+  command:
+    oc get
+       "istag/{{ utils_build_push_image_local_name }}:{{ utils_build_push_image_tag }}"
+       -n "{{ utils_build_push_image_namespace }}"
+  register: has_image
   failed_when: false
 
-- name: Create push pod and run image push
-  command: oc apply -f {{ artifact_extra_logs_dir }}/helper-image-push-pod.yml
+- name: Build the image
```

Comment:
Do we not still want to try building the helper image before attempting to build the actual image?

Reply:
I'm not sure any more how we discussed it, but from my POV, best is:
1. build the user image, it may take time and/or fail
2. if ^^^ is successful, and we want to push, build the helper image (should never fail)
3. if ^^^ is successful, and we want to push, run the Pod to push the image

I noticed you had 1 and 2 swapped, let me know if you see it differently!

Reply:
Ok yeah, given that 2 should never fail (and that pushing is optional), I think your order makes the most sense